File: backend/models/DoctorFeedback.js
const mongoose = require('mongoose');

const DoctorFeedbackSchema = new mongoose.Schema({
  doctorId: { type: String, required: true },
  patientId: { type: String, required: true },
  patientName: { type: String, required: true },
  rating: { type: Number, required: true, min: 1, max: 5 },
  feedback: { type: String, default: "" },
  appointmentId: { type: String, required: true },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('DoctorFeedback', DoctorFeedbackSchema);

File: app/src/main/java/com/example/medplus/model/DoctorFeedback.kt
package com.example.medplus.model

import com.google.firebase.Timestamp

/**
 * Data model for patient feedback on doctors.
 */
data class DoctorFeedback(
    val doctorId: String = "",
    val patientId: String = "",
    val rating: Int = 0,
    val feedback: String = "",
    val createdAt: Timestamp? = null
)
